k2/python/csrc/torch/fsa.cu

```diff
@@ -124,21 +233,21 @@ static void PybindDenseFsaVec(py::module &m) {
   // We do not need to access its members in Python
 
   // TODO(fangjun): add docstring for this funciton
-  pyclass.def(py::init(
-      [](torch::Tensor scores, torch::Tensor row_splits) -> DenseFsaVec * {
+  pyclass.def(
+      py::init([](torch::Tensor scores,
+                  torch::Tensor row_splits) -> std::unique_ptr<DenseFsaVec> {
```

Comment:
just a question, why is `row_splits` here instead of `RaggedShape`?

Reply:
`row_splits`, `sizes`, and `shape` are all fine here.
`RaggedShape` has not been wrapped to Python since I don't know when and where we need `RaggedShape` in Python, so `row_splits` is used here. `sizes` requires an extra call to `ExclusiveSum`.

Reply:
OK, thanks for the explanation. Just noted that when we pass `row_splits`, we may do a Device2Host copy in this line of this wraper function for `cached_tot_size`
https://github.com/k2-fsa/k2/blob/24afc76e23c2db4bd089f74d241ac5f82fe2fc5d/k2/python/csrc/torch/fsa.cu#L245

Reply:
If we pass a RaggedShape, there would also be a copy, but it happens in Python.